### ISSUE TITLE
Won't install on Ubuntu 16.04

### ISSUE THREAD

**Issue by gsportelli** [jupyterhub/the-littlest-jupyterhub#186]
I'm trying to install tljh in Ubuntu 16.04, but I get the following error:

    Checking if TLJH is already installed...
    TLJH already installed, upgrading...
    Upgrading TLJH installer...
    Setup tljh package
    Starting TLJH installer...
    Traceback (most recent call last):
      File "/usr/lib/python3.5/runpy.py", line 174, in _run_module_as_main
        mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
      File "/usr/lib/python3.5/runpy.py", line 144, in _get_module_details
        code = loader.get_code(mod_name)
      File "<frozen importlib._bootstrap_external>", line 767, in get_code
      File "<frozen importlib._bootstrap_external>", line 727, in source_to_code
      File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
      File "/opt/tljh/hub/lib/python3.5/site-packages/tljh/installer.py", line 326
        raise FileExistsError(f'/usr/bin/tljh-config exists but is not a symlink to {tljh_config_src}')
                                                                                                 ^
    SyntaxError: invalid syntax

I see that the offending line uses a string feature ('f' prefix) that is not available in python 3.5, which is the most recent python in Ubuntu 16.04.

**Comment by fomightez** [comment]
Per https://the-littlest-jupyterhub.readthedocs.io/en/latest/:

>"The Littlest JupyterHub (TLJH) can run on any server that is running at least Ubuntu 18.04. We have a bunch of tutorials to get you started."

Or...  
I had hit the same thing when I tried a couple of weeks ago before Amazon added a standard Ubuntu 18.04 to the `Quick-start` listing. I had considered forking and editing the f-string use out. You could try that. I decided against it figuring I'd probably hit another road block since clearly the documentation had been changed to move to Ubuntu 18 even though within the [bootstrap script](https://github.com/jupyterhub/the-littlest-jupyterhub/blob/master/bootstrap/bootstrap.py), it still references Ubuntu 16, specifically it reads, "Be compatible with Python 3.4 (since we support Ubuntu 16.04). Alternatively, there is a way to make f-strings work in older Python versions, see [here](https://stackoverflow.com/a/46182112/8508004).

I had also made a quick pass at updating Python on the Ubuntu 16.04-based cloud machine I was using but for some reason it continued to still use 3.5 because I probably had something not updated in a correct path. This another work-around you could consider if you cannot use Ubuntu 18.04.

**Comment by willingc** [comment]
Update the `README.md` and `docs/index.rst` to bold the requirement `Ubuntu 18.04`.

Search the docstrings of the source code and update reference to Ubuntu 16 to 18.04.

**Comment by olivierverdier** [comment]
I have managed to install TLJH on Ubuntu 16.04. The only thing you have to do is to start the bootstrapping with a more recent version than the one provided in Ubuntu. For instance, install Python using an anaconda distribution. Then, modify the `bootstrap.py` script so that it uses that Python executable instead. For instance by applying the patch here: https://github.com/olivierverdier/the-littlest-jupyterhub/commit/0be572514c1b08b6823eebbe7fce47a3abbf3389

In short, the only reason that 16.04 is not supported is that the bootstrap script uses the system python of Ubuntu. Use a more recent one instead, and everything works, as far as I can see.

**Comment by olivierverdier** [comment]
I have now made a full branch for installation on Ubuntu 16.04. There were a few more things to tweak to make it work completely. Here is the branch: https://github.com/olivierverdier/the-littlest-jupyterhub/tree/ubuntu-16-04
